# Clase #2 (script 2)

En este script se ejecutan procedimientos de carga de librerías, carga de datos y generación de tablas de frecuencias. Para este ejemplo se utilizará una base de datos llamada blood.txt la cual contiene información sobre la edad y presión sanguinea de 30 personas.

Se inicia con la carga de librerías.

In [1]:
import numpy as np
import scipy.stats as st
import pandas as pd
import os

Este script supone que el directorio donde se encuentra almacenado es C:. Si no es así, recuerde cambiar el valor de la variable path para que corresponda con la ubicación del script, y este se encuentre a la misma altura que la carpeta que la carpeta data. La carpeta data contiene la base de datos blood.txt.

In [2]:
os.getcwd()
path = "C:/Users/tic.tank/Documents/Miguel Orjuela/dcd_clase3"
os.chdir(path)
os.listdir(path)

['.ipynb_checkpoints', 'clase3_script1.ipynb', 'data', 'script1.py']

## Estimación puntual

Se carga el archivo `data_blood.txt` mediante el método read_csv de la librería `pandas`. También se generan medias numéricas de resumen con la función `describe()`.

In [3]:
filename = "data/data_blood.txt"
datos = pd.read_csv(filename, header = None, sep = "\s+", names = [u'Indice', u'Uno', u'Edad', u'Presión Sangre'])
print(datos.describe())

          Indice   Uno       Edad  Presión Sangre
count  30.000000  30.0  30.000000       30.000000
mean   15.500000   1.0  45.133333      142.533333
std     8.803408   0.0  15.294203       22.581245
min     1.000000   1.0  17.000000      110.000000
25%     8.250000   1.0  36.750000      125.750000
50%    15.500000   1.0  45.500000      141.000000
75%    22.750000   1.0  56.000000      157.000000
max    30.000000   1.0  69.000000      220.000000


A partir del dataset `datos` se genera la serie `datosPresion` y se generan descriptivos únicamente para esta variable.

In [4]:
datosPresion = datos[u'Presión Sangre']
ser = pd.Series(datosPresion)
print(ser.describe())

count     30.000000
mean     142.533333
std       22.581245
min      110.000000
25%      125.750000
50%      141.000000
75%      157.000000
max      220.000000
Name: Presión Sangre, dtype: float64


In [5]:
n = len(datosPresion)
print("n: ", n)
print("media: ", np.mean(datosPresion))
print("desv estandar (n) : ", np.std(datosPresion))
print("desv estandar (n-1) : ", np.std(datosPresion, ddof = 1))

n:  30
media:  142.53333333333333
desv estandar (n) :  22.201701636486234
desv estandar (n-1) :  22.581245396969603


## Estimación por intervalos de confianza

Se utiliza el método `t.interval()` de la librería `scipy.stats`. De acuerdo a la documentación del método, el primer parámetro es la confianza, o la probabilidad deseada de que la media caiga en ese intervalo; `len(datosPresion)-1` define los grados de libertad de la distribución; `loc` indica la media muestral, a la cual asignamos el valor de `np.mean(datosPresion)`); y `scale` es el parámetro del error estándar de la media, calculado con el método `st.sem(datosPresion)`.

Como observación adicional, es importante recordar que el error estándar de la media se calcula de la siguiente forma:
$$SEM = \frac{\sigma}{\sqrt{n}}$$


In [6]:
import numpy as np
import scipy.stats as st

intervalo = st.t.interval(0.95, len(datosPresion)-1,loc = np.mean(datosPresion), scale = st.sem(datosPresion))
print(intervalo)

(134.1013577264643, 150.96530894020236)


El resultado obtenido indica que con una probabilidad del 95%, la la media poblacional de la presión arterial se encuentra entre los valores 134.1013577264643 y 150.96530894020236.

Una alternativa para calcular este intervalo es mediante la librería `statsmodels.stats.api`, como se ilustra a continuación.

In [7]:
import statsmodels.stats.api as sms
intervalo = sms.DescrStatsW(datosPresion).tconfint_mean(0.05)
print(intervalo)

(134.10135772646427, 150.9653089402024)


C:\Users\tic.tank\AppData\Local\Continuum\anaconda3\lib\site-packages\statsmodels\compat\pandas.py:56: FutureWarning: The pandas.core.datetools module is deprecated and will be removed in a future version. Please use the pandas.tseries module instead.
  from pandas.core import datetools
